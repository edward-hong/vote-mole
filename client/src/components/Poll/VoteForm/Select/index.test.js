import React from 'react'
import { render } from '@testing-library/react'

import Select from '.'

const pollOptions = [
	{
		votes: 3,
		option: 'English',
	},
	{
		votes: 6,
		option: 'French',
	},
	{
		votes: 9,
		option: 'German',
	},
	{
		votes: 5,
		option: 'Russian',
	},
]

describe('Select', () => {
	it('shows error message when there is an error', () => {
		const error = 'Please make a selection'
		const { getByText } = render(
			<Select
				input={{ name: 'select' }}
				meta={{ touched: true, error }}
				pollOptions={pollOptions}
				type="select"
			/>,
		)

		expect(getByText(error)).toBeInTheDocument()
	})

	it('renders correctly when valid', () => {
		const { container } = render(
			<Select
				input={{ name: 'select' }}
				meta={{ touched: true, error: false }}
				pollOptions={pollOptions}
				type="select"
			/>,
		)

		expect(container.firstChild).toMatchSnapshot()
	})

	it('renders correctly when untouched', () => {
		const { container } = render(
			<Select
				input={{ name: 'select' }}
				meta={{ touched: false, error: false }}
				pollOptions={pollOptions}
				type="select"
			/>,
		)

		expect(container.firstChild).toMatchSnapshot()
	})
})
