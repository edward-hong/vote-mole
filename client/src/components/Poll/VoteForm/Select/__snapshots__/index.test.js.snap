// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly when untouched 1`] = `
<div
  class="form-group"
>
  <label
    class=""
    for="select"
  />
  <select
    class="form-control"
    name="select"
  >
    <option
      disabled=""
    />
    <option>
      English
    </option>
    <option>
      French
    </option>
    <option>
      German
    </option>
    <option>
      Russian
    </option>
  </select>
</div>
`;

exports[`Select renders correctly when valid 1`] = `
<div
  class="form-group"
>
  <label
    class=""
    for="select"
  />
  <select
    class="is-valid form-control"
    name="select"
  >
    <option
      disabled=""
    />
    <option>
      English
    </option>
    <option>
      French
    </option>
    <option>
      German
    </option>
    <option>
      Russian
    </option>
  </select>
</div>
`;
